include/common/common.h

```diff
@@ -225,4 +225,44 @@ template <typename T> std::string type_to_string() {
         return "bf16";
 
     return "unknown";
-}
\ No newline at end of file
+}
+
+template <typename T> std::vector<T> vertical_pack(std::vector<T> &matrix, int rows, int cols) {
```

Comment:
we have the 'packed' function in devicematrices.h which does the same thing. Can we remove the redundancy? I.e. one function host and device which works on T * const arrays and then all variations call that one fucntion?

Reply:
move to result_check or io
remove redundancies in io/common